### PR TITLE
hfr delaminations now explains whats wrong

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Adds radio messages to the delaminating HFR to let atmosians know what's wrong with it

## Why It's Good For The Game
Less obscure reason for delamination and helps players understand what to fix when something goes wrong. Unless the HFR has been emped

## Changelog

:cl:
qol: Adds radio messages to the delaminating HFR to let atmosians know what's wrong with it. Unless the HFR has been emped
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/fusion/hfr_core.dm [hunk 1]
```diff
@@ -159,6 +159,12 @@
 	///Var used in the meltdown phase
 	var/final_countdown = FALSE
 
+	var/high_power_damage = FALSE
+	var/high_fuel_mix_mole = FALSE
+	var/iron_content_damage = FALSE
+	var/iron_content_increasing = FALSE
+	var/emped = FALSE
```
Comment (MrMelbert) [inline code/modules/atmospherics/machinery/components/fusion/hfr_core.dm:166]: Document these

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/components/fusion/hfr_core.dm [hunk 2]
```diff
@@ -159,6 +159,17 @@
...
+	///Damage caused because of the high power tier
+	var/high_power_damage = FALSE
+	///Damage caused by too much fuel gas inside
+	var/high_fuel_mix_mole = FALSE
+	///Damage caused by iron content
+	var/iron_content_damage = FALSE
+	///Check if iron inside the core is increasing
+	var/iron_content_increasing = FALSE
+	///If this is true only a non readable message will pop out
+	var/emped = FALSE
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/fusion/hfr_core.dm:171]: You should turn these into bitflags, I get the feeling it'll make the code slightly nicer

File: code/modules/atmospherics/machinery/components/fusion/hfr_core.dm [hunk 3]
```diff
@@ -159,6 +159,9 @@
 	///Var used in the meltdown phase
 	var/final_countdown = FALSE
 
+	///Warning damage flags
+	var/warning_damage_flags = NONE
```
Comment (MrMelbert) [inline code/modules/atmospherics/machinery/components/fusion/hfr_core.dm:163]: Sorry but this doc comment is pretty much useless. It should at least mention what the flags are used for, and what are potential flags.

File: code/modules/atmospherics/machinery/components/fusion/hfr_core.dm [hunk 4]
```diff
@@ -159,6 +159,9 @@
 	///Var used in the meltdown phase
 	var/final_countdown = FALSE
 
+	///Warning damage flags
```
Comment (Ghilker) [inline code/modules/atmospherics/machinery/components/fusion/hfr_core.dm:162]:
```suggestion
	///Flags used in the alert proc to select what messages to show when the HFR is delaminating (HYPERTORUS_FLAG_HIGH_POWER_DAMAGE | HYPERTORUS_FLAG_HIGH_FUEL_MIX_MOLE | HYPERTORUS_FLAG_IRON_CONTENT_DAMAGE | HYPERTORUS_FLAG_IRON_CONTENT_INCREASE | HYPERTORUS_FLAG_EMPED)
```

**Pull request merged**